src/wrapper/signature.rs

```diff
@@ -79,15 +78,17 @@ pub struct TypeSignature {
 
 impl TypeSignature {
     /// Parse a signature string into a TypeSignature enum.
-    pub fn from_str<S: AsRef<str>>(s: S) -> Result<TypeSignature> {
-        Ok(match parser(parse_sig)
-            .parse(State::new(s.as_ref()))
-            .map(|res| res.0)
-        {
-            Ok(JavaType::Method(sig)) => *sig,
-            Err(e) => return Err(format_error_message(&e, s.as_ref()).into()),
-            _ => unreachable!(),
-        })
+    pub fn from_string<S: AsRef<str>>(s: S) -> Result<TypeSignature> {
```

Comment:
It is a breaking change — why is it needed (is it?)?

Reply:
Yeap... Clippy suggested implementing `FromStr` (not in our case because of arguments type) or renaming it. Perhaps, it's better to just suppress it. 🤔